### PR TITLE
ci: add a deployment pipeline to fly.io

### LINKED ISSUE #20: Creation of the deployment pipeline

### PR DESCRIPTION
**Pull request by Harian-Elyoth**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+RUN apt-get update -qq && \
+    apt-get install --no-install-recommends -y build-essential node-gyp openssl pkg-config python-is-python3
+
+# Install node modules
+COPY --link package-lock.json package.json ./
+RUN npm ci --include=dev
+
+# Generate Prisma Client
+COPY --link prisma .
+RUN npx prisma generate
```
Comment (hcanon42) [inline Dockerfile:29]: shouldn't we run ```npx prisma migrate dev```(or ```npm run db:migrate```) ?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# Set production environment
+ENV NODE_ENV="production"
+
+
+# Throw-away build stage to reduce size of final image
+FROM base as build
+
+# Install packages needed to build node modules
+RUN apt-get update -qq && \
+    apt-get install --no-install-recommends -y build-essential node-gyp openssl pkg-config python-is-python3
```
Comment (hcanon42) [inline Dockerfile:21]: Just a question: why do we need python-is-python3 ?

**Review by hcanon42 — COMMENTED** [review]
Just two small questions

File: Dockerfile (hunk 1 above)
Comment (Harian-Elyoth) [reply Dockerfile:29]: I add the npx prisma migrate deploy (because it's for production)

File: Dockerfile (hunk 2 above)
Comment (Harian-Elyoth) [reply Dockerfile:21]: Auto generated file by Fly.io They might need it.

**Review by hcanon42 — APPROVED** [review]
Seems good, I didn't run it but I trust you did

**Pull request merged**